src/main/java/io/stargate/sgv2/jsonapi/service/resolver/update/TableUpdatePushResolver.java

```diff
@@ -0,0 +1,307 @@
+   */
+  @Override
+  public List<ColumnAssignment> resolve(TableSchemaObject table, ObjectNode arguments) {
+    return arguments.properties().stream()
+        .map(
+            entry -> {
+              var column = entry.getKey();
+              var apiColumnDef = checkUpdateColumnExists(table, column);
+              var inputValue = entry.getValue();
+
+              // $push only works for map/set/list column
+              if (!apiColumnDef.type().apiSupport().update().push()) {
+                throw UpdateException.Code.UNSUPPORTED_UPDATE_OPERATOR.get(
+                    errVars(
+                        table,
+                        map -> {
+                          map.put("operator", "$push");
+                          map.put("targetColumn", errFmt(apiColumnDef.name()));
+                        }));
+              }
+
+              JsonLiteral<?> shreddedValue = null;
+              // resolve $push value for set/list column
+              if (apiColumnDef.type().typeName() == ApiTypeName.SET
+                  || apiColumnDef.type().typeName() == ApiTypeName.LIST) {
+                shreddedValue = resolvePushForListSet(table, inputValue);
+              }
+
+              // resolve $push value for map column
+              if (apiColumnDef.type().typeName() == ApiTypeName.MAP) {
+                shreddedValue = resolvePushForMap(table, inputValue);
+              }
+
+              return new ColumnAssignment(
+                  new ColumnAppendToAssignment(),
+                  table.tableMetadata(),
+                  CqlIdentifierUtil.cqlIdentifierFromUserInput(column),
+                  shreddedValue);
+            })
+        .toList();
+  }
+
+  /**
+   * Resolve $push operator value for list/set column.
+   *
+   * <p>Example(Push single element to the list/set):
+   *
+   * <ul>
+   *   <li>list. <code>{"$push" : {"textList" : "textValue", "intList" : 111}}</code>
+   *   <li>set. <code>{"$push" : {"textSet" : "textValue", "intSet" : 111}}</code>
+   * </ul>
+   *
+   * <p>Example(Push multiple elements to the list/set):
+   *
+   * <ul>
+   *   <li>list. <code>
+   *       {"$push" : {"textList" : {"$each": ["textValue1", "textValue2"]}, "intList" : {"$each": [1,2]}}}
+   *       </code>
+   *   <li>set. <code>
+   *       {"$push" : {"textSet" : {"$each": ["textValue1", "textValue2"]}, "intSet" : {"$each": [1,2]}}}
+   *       </code>
+   * </ul>
+   *
+   * TODO $position for list column
+   *
+   * @param table TableSchemaObject
+   * @param inputValue jsonNode value for the $push column, E.G. if the operator is {"$push" :
+   *     {"textList" : {"$each": ["textValue1", "textValue2"]}}, then inputValue is {"$each":
+   *     ["textValue1", "textValue2"]}
+   * @return JsonLiteral
+   */
+  private JsonLiteral<?> resolvePushForListSet(TableSchemaObject table, JsonNode inputValue) {
+    JsonLiteral<?> shreddedValue;
+
+    switch (inputValue.getNodeType()) {
```

Comment:
nit: not sure this is the best use of a switch statement, may be easier as some if's and returns because nothing happens after the switch